### ISSUE TITLE
type 'typing.TypeVar' is not an acceptable base type

### ISSUE THREAD

**Issue by crazydef** [Azure/azure-sdk-for-python#33442]
- **Package Name**: azure-storage-blob azure-identity
- **Package Version**: azure-core 1.26.4, azure-identity 1.12.0, azure-storage-blob 12.16.0
- **Operating System**: Windows
- **Python Version**: 3.12.0

**Describe the bug**
We have a python script running as part of our build system  It's been running fine for the better part of a year (both locally on developer machines and on DevOps under Windows, Linux, and MacOS).

This script started breaking a few days ago for one member of our team after they upgraded to Python 3.12.  (At least, we think that's the reason.)

**To Reproduce**
The part of the script that breaks is as follows:

```
from azure.storage.blob import BlobServiceClient, ContainerClient
```

Under Python 3.12 this throws an exception with the following suitably cryptic message:
```
type 'typing.TypeVar' is not an acceptable base type
```

**Expected behavior**
Under Python 3.11.3 (the version I am using on Windows) the packages are imported as expected.  On DevOps running Python 3.9.13, this script also works as expected.

**Comment by xiangyan99** [comment]
Thanks for the feedback, we’ll investigate asap.

**Comment by vincenttran-msft** [comment]
Hi @crazydef James, thanks for your inquiry. From a cursory search, it seems that this is a known issue with Python 3.12+ and certain versions of `typing_extensions` (seems to be 4.4.0 and 4.5.0):
https://github.com/python/typing_extensions/issues/243

With that being said, to verify this is the root cause of the issue, can you please check what installed version of `typing_extensions` you are using? If it is one of versions stated above, then simply updating to the latest version (or at the minimum 4.6.0), should resolve your issue.

Please let me know if this works for you and seems to be the cause! As a follow up, we will look to possibly bump up our `typing_extensions` version when possible if this does seem to be the case.

Thanks!

**Comment by crazydef** [comment]
Hi @vincenttran-msft.

Unfortunately the person in question had to roll back his python installation to continue working.

However, it turns out another person in our team has been successfully using Python 3.12 for quite some time with no issues.  He is using typing_extensions version 4.8.0.

I have just upgraded my Python to 3.12 and I get typing_extensions version 4.9.0 by default.  So I'm not sure what was going on with this other broken machine.  Probably a messed up Python install more than anything else.

**Comment by jalauzon-msft** [comment]
Hi @crazydef James, I have confirmed the issue to be the bug in `typing-extensions==4.4.0-4.5.0` and Python 3.12 so anybody running into this issue should go ahead an update their `typing-extensions` dependency which should resolve the issue. What likely happened to the person who first encountered the issue is they updated their Python version but did not update any of the package versions in their environment. If they wish to use Python 3.12, they can try again but be sure to update their `typing-extensions` version (`pip install typing-extensions --upgrade`). However, it may be better to just start with a clean environment and install everything again to make sure they get the latest for everything. I do also want to just note we have not released official support for Python 3.12 yet but plan to in our next release.

### Additional details on how this can occur

As our dependencies stand today, it is still possible for new users to get into this scenario but unlikely. Our dependency `typing-extensions>=4.3.0` can allow for broken versions but `azure-core` (also our dependency) has their dependency set to `typing-extensions>=4.6.0` meaning recent versions of that will not allow the broken version to be installed. However, our dependency on core is quite old, so it's possible to install an older versions of `azure-core` as well which would allow for a broken version of `typing-extensions`. All that is to say, any new installs of the package should almost definitely not be seeing this issue but if someone upgrades just `azure-storage-blob` without updating dependencies its possible. So, I will put out a PR soon that bumps our dependency on `typing-extensions` to `>=4.6.0` so that in our next release we guarantee this error will not occur on the latest version of blob.

**Comment by crazydef** [comment]
OK, good to know.  Thanks for the update.
